pkg/webhook/connection/connection_webhook.go

```diff
@@ -142,6 +142,37 @@ func (w *Webhook) ValidateDelete(ctx context.Context, obj runtime.Object) (admis
 	if len(extAttachments.Items) > 0 {
 		return nil, errors.Errorf("connection has external attachments")
 	}
+	if conn.Spec.Management != nil {
+		mgmtConnList := &wiringapi.ConnectionList{}
+		if err := w.Client.List(ctx, mgmtConnList, client.MatchingLabels{
+			wiringapi.LabelConnectionType: wiringapi.ConnectionTypeManagement,
+		}); err != nil {
+			return nil, errors.Errorf("error listing servers on management connection")
+		}
+		if len(mgmtConnList.Items) <= 1 {
+			return nil, errors.Errorf("management connection is last one and cannot be deleted")
+		}
+	}
+
+	// This is a light check to make sure that no mclags links transit this domain before we delete the domain
+	if conn.Spec.MCLAGDomain != nil {
+		labels := conn.Spec.ConnectionLabels()
+		searchLabels := make(client.MatchingLabels)
```

Comment:
you don't need to create a copy as .ConnectionLabels() already returns a new map

Reply:
Thank you